PNI styling updates

mmmavis opened the pull request:
Closes #3732

Edit: https://foundation-mofostaging-pr-3788.herokuapp.com/cms/buyersguide/product/edit/64/
Page: https://foundation-mofostaging-pr-3788.herokuapp.com/en/privacynotincluded/products/vivofit-jr-2/

Notes: there are some font weight inconsistencies in the Sketch file... I'm not sure if those were made intentionally. I altered a few in this PR for the sake of consistency but let me know I should just follow what's in the Sketch file.

Pomax commented:
Parental controls and Biomertric data hiding on "not applicable" was done in #3733

Pomax commented:
From Slack: the star CSS needs a tweak so that value-yes = black, value-no/value-u = grey, and value-na = half (the scss I wrote maps yes to black, no to grey, unknown to half, and nothing for not applicable)

kristinashu commented:
Sorry, when I put real content in I broke the link to the test page - it's here now https://foundation-mofostaging-pr-3788.herokuapp.com/en/privacynotincluded/products/vivofit-jr-2/

This is great! I don't think there are any critical blockers so I'll approve the PR. 

There are a few styling things, mostly around vertical spacing but again, there are not critical and can be done as a follow up.
- update `Minimum Security Standards` text to .body-large (it was something in between body small and large in the sketch but I think we can just do large)
- stars need a little more space between eachother
-  in general, more vertical space is needed between the diving lines and the text
-  in general, text should be vertically centered between the diving lines even when there is no help text
-  add diving line below `User friendly privacy info?` and 😮

Here's a dodgy comp trying to show some of these:

![Screen-Shot-2019-10-16-at-6 27 56-PM](https://user-images.githubusercontent.com/1068783/66970999-58d04f80-f044-11e9-82f2-da965202e462.jpg)

@beccaklam could you please take a look at this as well and add any issues you see? If you find something, please point out what are important issue and what is a nice to have.

kristinashu commented:
Also, this section looks so good! Love the gradient.

<img width="877" alt="image" src="https://user-images.githubusercontent.com/1068783/66971549-959d4600-f046-11e9-803e-45763b848724.png">

Pomax commented on network-api/networkapi/buyersguide/templates/product_page.html:
```diff
@@ -73,53 +79,44 @@ <h2 class="h1-heading col-md-10">{{product.name}}</h2>
 
-        <div class="worst-case">
-          <h3 class="h3-heading"><span class="creep-face-emoji">😮</span>What could happen if something went wrong</h3>
+        <div class="worst-case-box px-4 px-md-5 pt-5 pb-2 pb-md-4 mb-5 text-center">
```
now _that's_ a bootstrap class

Pomax commented on source/sass/buyers-guide/views/product.scss:
```diff
@@ -89,13 +92,32 @@ $product-detail-padding-x: (
+  .worst-case-box {
+    position: relative;
+    border-width: 1px;
+    border-image: linear-gradient(to left, $pni-pink 0%, $pni-blue 100%) 1 / auto;
```
it looks like Chrome doesn't like this, but if I add the `-webkit-` prefix it looks like Chrome forgets what a border is, and just makes it the entire background O_O

![image](https://user-images.githubusercontent.com/177243/67021639-c9af5000-f0b4-11e9-8e15-abdd6fd8b1d1.png)

Pomax replied:
The rule specified in https://css-tricks.com/gradient-borders-in-css/ does seem to work in both Chrome and Firefox

mmmavis replied:
Thanks for the link @Pomax . It turned out just adding `border-style: solid` fixed the Chrome issue. Phew! One fewer divs to maintain 😂

beccaklam commented:
@mmmavis The page looks great! +1 to @kristinashu's changes and aside from +1 @Pomax 's find above ☝️ with the missing gradient in chrome, I found two little things:
- [ ] We're missing an hr under _User friendly privacy info?_ section:
### PR:
![Screen Shot 2019-10-17 at 11 05 19 AM](https://user-images.githubusercontent.com/20977376/67021697-063b7580-f0ce-11e9-8ee2-15e9fca14b5f.png)
### Comp:
![Screen Shot 2019-10-17 at 10 59 20 AM](https://user-images.githubusercontent.com/20977376/67021711-0cc9ed00-f0ce-11e9-8e00-d08e162eed9f.png)
I don't know if it's completely necessary but I feel like it "closes off" the section better. @kristinashu Thoughts?
- [ ] Also for the Minimum Security Standards section, there's still some spacing that's showing up under the section title even when there's no elaboration text. Would it be possible to get ride of that?
### PR:
![Screen Shot 2019-10-17 at 11 02 50 AM](https://user-images.githubusercontent.com/20977376/67021836-3f73e580-f0ce-11e9-8b55-93288ec01cb9.png)

mmmavis commented:
Updated the PR with the following. Gonna leave other styling tweaks mentioned above as follow-up.

- fixed gradient border bug on Chrome
- add divider line around the `User friendly privacy info?` section
- update `Minimum Security Standards` text to .body-large
- increase spacing in between the stars

kristinashu commented:
Nice! Those are looking good!

Pomax commented:
Sweet!

kristinashu commented:
I've opened a ticket for the remaining styling tweaks https://github.com/mozilla/foundation.mozilla.org/issues/3793

Hopefully this is it but I would like to review it again and more thoroughly once it's on staging.

beccaklam commented:
> Nicely spotted @Pomax!
> 
> And thank you @beccaklam! I did mention those things but definitely not as clearly or nicely as you have. @mmmavis let me know if you need clarification on any of these.

Lol oops 😅 Definitely skim-read that too quickly.
